src/main/io/gps.c

```diff
@@ -1805,12 +2350,7 @@ static bool gpsNewFrameUBLOX(uint8_t data)
             _step++;
             _ck_b += (_ck_a += data);       // checksum byte
             _payload_length += (uint16_t)(data << 8);
-#if DEBUG_UBLOX_FRAMES
-    debug[3] = _payload_length;
-#endif
-            if (_payload_length > UBLOX_PAYLOAD_SIZE) {
-                _skip_packet = true;
-            }
+
```

Comment:
limiting payload length here to
`_payload_length = (_payload_length <= UBLOX_PAYLOAD_SIZE) ? _payload_length : UBLOX_PAYLOAD_SIZE;`
would avoid double `if`s in step 6, so it would become
```
    _ck_b += (_ck_a += data);       // checksum byte
    if (_payload_counter++ < _payload_length) {
        _buffer.bytes[_payload_counter] = data;
    } else {
        _step++;
    }
```
overall yielding a 7-10% CPU reduction in GPU message processing (step 6 is by far the most frequently called function, so any optimisation there is win)

Reply:
I tried this out, I think correctly, but it broke the connection process.

Reply:
Edited the code, the infamous off by one error. Should work now

Reply:
OK that definitely works now - thanks - maybe @ledvinap could give us his thoughts?